tests/components/openweathermap/test_weather.py

```diff
@@ -0,0 +1,132 @@
+"""Test the OpenWeatherMap weather entity."""
+
+from unittest.mock import AsyncMock, patch
+
+import pytest
+
+from homeassistant.components.openweathermap.const import (
+    ATTR_API_MINUTE_FORECAST,
+    DOMAIN,
+    OWM_MODE_V25,
+    OWM_MODE_V30,
+)
+from homeassistant.components.openweathermap.weather import SERVICE_GET_MINUTE_FORECAST
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_MODE,
+    CONF_NAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def mock_config_entry():
+    """Create a mock OpenWeatherMap config entry."""
+    return MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_API_KEY: "test_api_key",
+            CONF_LATITUDE: 12.34,
+            CONF_LONGITUDE: 56.78,
+            CONF_MODE: OWM_MODE_V30,
+            CONF_NAME: "OpenWeatherMap",
+        },
+        unique_id="test_unique_id",
+    )
+
+
+@pytest.fixture
+def mock_weather_data():
+    """Return mock weather data."""
+    return {
+        ATTR_API_MINUTE_FORECAST: [
+            {"time": "2024-10-01T12:00:00Z", "precipitation": 0},
+            {"time": "2024-10-01T12:01:00Z", "precipitation": 0.1},
+            {"time": "2024-10-01T12:02:00Z", "precipitation": 0.23},
+            {"time": "2024-10-01T12:03:00Z", "precipitation": 0},
+        ]
+    }
+
+
+@pytest.fixture(autouse=True)
+def mock_weather_update():
+    """Mock the WeatherUpdateCoordinator to prevent API calls."""
+    with patch(
+        "homeassistant.components.openweathermap.coordinator.WeatherUpdateCoordinator._async_update_data",
```

Comment:
We should not patch internals, we should only patch the external library.

Reply:
Good point. Patching is now limited to the HTTP call with the data provided by a static JSON file. This has had the added benefit of increasing the test coverage.